services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/liitteet/AttachmentServiceImpl.kt

```diff
@@ -0,0 +1,96 @@
+package fi.hel.haitaton.hanke.liitteet
+
+import fi.hel.haitaton.hanke.HankeRepository
+import fi.hel.haitaton.hanke.currentUserId
+import fi.hel.haitaton.hanke.getCurrentTimeUTCAsLocalTime
+import java.util.*
+import javax.persistence.EntityNotFoundException
+import org.springframework.orm.jpa.JpaObjectRetrievalFailureException
+import org.springframework.transaction.annotation.Transactional
+import org.springframework.web.multipart.MultipartFile
+
+private val supportedFiletypes =
+    setOf(
+        "application/pdf",
+        "application/msword",
+        "application/vnd.openxmlformats-officedocument.wordprocessingml.document",
+        "image/jpeg",
+        "image/png",
+        "image/vnd.dwg"
+    )
+
+private const val FILESIZE_MAXIMUM_MEGABYTES = 10
+
+class AttachmentServiceImpl(
+    val hankeRepository: HankeRepository,
+    val attachmentRepository: AttachmentRepository,
+) : AttachmentService {
+    @Transactional
+    override fun add(hankeTunnus: String, liite: MultipartFile): HankeAttachment {
+        if (!supportedFiletypes.contains(liite.contentType)) {
+            throw AttachmentUploadError("Filetype not supported: " + liite.contentType)
+        }
+
+        if (liite.size > 1024 * 1024 * FILESIZE_MAXIMUM_MEGABYTES) {
+            throw AttachmentUploadError(
+                "File size should not exceed ${FILESIZE_MAXIMUM_MEGABYTES}Mb"
```

Comment:
b is bit, B is byte.
```suggestion
                "File size should not exceed ${FILESIZE_MAXIMUM_MEGABYTES}MB"
```